server/src/domain/metadata/metadata.service.ts

```diff
@@ -156,9 +157,18 @@ export class MetadataService {
     await this.applyMotionPhotos(asset, tags);
     await this.applyReverseGeocoding(asset, exifData);
     await this.assetRepository.upsertExif(exifData);
+    let localDateTime = exifData.dateTimeOriginal ?? undefined;
+
+    const dateTimeOriginal = exifDate(firstDateTime(tags as Tags)) ?? exifData.dateTimeOriginal;
+    const timeZoneOffset = tzOffset(firstDateTime(tags as Tags)) ?? 0;
+
+    if (dateTimeOriginal && timeZoneOffset) {
+      localDateTime = new Date(dateTimeOriginal.getTime() + timeZoneOffset * 60000);
```

Comment:
This looks good. Doing it with luxon probably would be cleaner, but it's probably good enough.

Reply:
Maybe I'll use luxon later on